Module2_HW2/Module2_HW2/Store.cs

```diff
@@ -0,0 +1,239 @@
+﻿using Module2_HW2.Entity;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Module2_HW2
+{
+    public class Store
+    {
+        private ShopingCart _shoppingCart = new ShopingCart();
+        private List<Product> _listProducts = new List<Product>();
+
+        public Store() 
+        {
+            AddDataToStorage();
+        }
+
+        public int FindIndexInStorage(Product product)
+        {
+            for (int i = 0; i < _listProducts.Count(); i++)
+            {
+                if (_listProducts[i].Name == product.Name)
+                {
+                    return i;
+                }
+            }
+            return -100;
+        }
+        public void AddToStorage(Product product)
+        {
+            int index = FindIndexInStorage(product);
+            if (index >= 0)
+            {
+                _listProducts[index].Amount = _listProducts[index].Amount + product.Amount;
+            }
+            else 
+            {
+                _listProducts.Add(product);
+            }
+        }
+
+        public bool RemoveFromStorage(Product product)
+        {
+            int index = FindIndexInStorage(product);
+            if (index >= 0)
+            {
+                if (_listProducts[index].Amount < product.Amount)
+                {
+                    Console.WriteLine("You can't remove this product. Please  review store");
+                    return false;
+                }
+                _listProducts[index].Amount = _listProducts[index].Amount - product.Amount;
+                return true;
+            }
+            else
+            {
+                Console.WriteLine("You can't remove this product. Please  review store");
+                return false;
+            }
+        }
+
+        public bool ReviewStore() 
+        {
+            if (_listProducts.Count == 0) 
+            {
+                return false;
+            }
+            foreach (var item in _listProducts) 
+            {
+                Console.WriteLine($"Name {item.Name}, Amount {item.Amount}");
+            }
+            return true;
+        }
+
+        public bool AddToCart(Product products)
+        {
+            bool flag =_shoppingCart.AddToCart(products);
+            return flag;
+        }
+
+        public bool RemoveFromCart(Product products)
+        {
+            bool flag = _shoppingCart.RemoveFromCart(products);
+            return flag;
+        }
+
+        public void CartConfirmation()
+        {
+            _shoppingCart.CartConfirmation();
+        }
+
+        public void ReviewShopingCart()
+        {
+            _shoppingCart.DisplayShopingCart();
+        }
+
+        private void AddDataToStorage()
+        {
+            Product product1 = new Product("keyboard", 10);
```

Comment:
Why initialization with data is a part of that class? What happens if I want to change the initialization list (add other products)?